### PR TITLE
3.38.0: Adds new SDK versions and contract files

### PR DESCRIPTION
**Pull request by kundadebdatta**
# Pull Request Template

## Description

This bumps the SDK GA version from `3.37.1` to `3.38.0` and preview version from `3.37.1-preview` to `3.39.0-preview.0`. Please see the below changes in the GA public contract.

image

image

Also highlighted below are the preview contract changes.

image

_[Note: To understand about the GA and preview versions styles, please take a look at our [versioning guidelines and principles](https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/docs/versioning.md)]_

### CODE REVIEW

**Review by NaluTripician — APPROVED** [review]

**Review by sourabh1007 — APPROVED** [review]

**Review by NaluTripician — APPROVED** [review]

**Comment by kirankumarkolli** [comment]
ODE naming following up with query team.

**Review by kirankumarkolli — DISMISSED** [review]

File: changelog.md [hunk 1]
```diff
@@ -13,6 +13,29 @@ Preview features are treated as a separate branch and will not be included in th
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+### <a name="3.39.0-preview.0"/> [3.39.0-preview.0](https://www.nuget.org/packages/Microsoft.Azure.Cosmos/3.39.0-preview.0) - 2024-01-29
+
+#### Added
+- [4138](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4138) Query: Adds support for System.Text LINQ Custom Serializer (#4138)
```
Comment (ealsur) [inline changelog.md:19]: I would not say this is for STJ but rather:

```suggestion
- [4138](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4138) Query: Adds support for LINQ Custom Serializer (#4138)
```

File: changelog.md [hunk 2]
```diff
@@ -13,6 +13,29 @@ Preview features are treated as a separate branch and will not be included in th
...
+
+### <a name="3.38.0"/> [3.38.0](https://www.nuget.org/packages/Microsoft.Azure.Cosmos/3.38.0) - 2024-01-29
+
+#### Fixed
+- [4205](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4205) Per Partition Automatic Failover: Fixes Metadata Requests Retry Policy (#4205)
```
Comment (ealsur) [inline changelog.md:24]: This PPAF public?

File: changelog.md [hunk 3]
```diff
@@ -13,6 +13,29 @@ Preview features are treated as a separate branch and will not be included in th
...
+- [4229](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4229) GatewayClientStore: Fixes an issue with dealing with invalid JSON HTTP responses (#4229)
+- [4260](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4260) Query: Fixes LINQ Translation of SqlNullLiteral Values (#4260)
+- [4276](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4276) Change Feed Processor: Fixes LeaseLostException on Notifications API for Renewer (#4276)
+
+#### Added
+- [4122](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4122) Query: Adds Optimistic Direct Execution configuration override support on the Client (#4122)
+- [4218](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4218) Distributed Tracing: Refactors Samples to use latest cosmosdb SDK (#4218)
```
Comment (ealsur) [inline changelog.md:31]: Remove this one - this is a sample update

File: changelog.md [hunk 4]
```diff
@@ -13,6 +13,29 @@ Preview features are treated as a separate branch and will not be included in th
...
+- [4220](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4220) Change Feed Processor: Fixes disposal of unused CancellationTokenSource (#4220)
```
Comment (ealsur) [inline changelog.md:32]: Move this one to fixes

Comment (kundadebdatta) [reply changelog.md]: Done.

File: changelog.md (hunk 3 above)
Comment (kundadebdatta) [reply changelog.md]: Done.

File: changelog.md (hunk 2 above)
Comment (kundadebdatta) [reply changelog.md]: The PPAF terminology is not, however the goal is to make it publicly available. Therefore, I have updated the text to reflect that.

File: changelog.md (hunk 1 above)
Comment (kundadebdatta) [reply changelog.md]: Done.

File: changelog.md [hunk 5]
```diff
@@ -13,6 +13,29 @@ Preview features are treated as a separate branch and will not be included in th
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+### <a name="3.39.0-preview.0"/> [3.39.0-preview.0](https://www.nuget.org/packages/Microsoft.Azure.Cosmos/3.39.0-preview.0) - 2024-01-29
+
+#### Added
+- [4138](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4138) Query: Adds support for System.Text LINQ Custom Serializer (#4138)
+
+### <a name="3.38.0"/> [3.38.0](https://www.nuget.org/packages/Microsoft.Azure.Cosmos/3.38.0) - 2024-01-29
```
[the remark this answers is not in the document]
Comment (kundadebdatta) [reply changelog.md]: Captured the detailed description.

File: changelog.md [hunk 6]
```diff
@@ -13,6 +13,38 @@ Preview features are treated as a separate branch and will not be included in th
...
+- [4241](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4241) GlobalEndpointManager: Fixes Unobserved and Unhandled Exception from Getting Thrown (#4241)
+
+#### Added
+- [4122](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4122) Query: Adds Optimistic Direct Execution configuration override support on the Client (#4122)
+- [4240](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4240) BulkMode: Adds PartitionKeyRangeId in Bulk Mode and TransactionalBatch Response Headers (#4240)
+- [4252](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4252) Query: Adds Request Charge to Query Metrics (#4252)
+- [4225](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4225) Query: Refactors Optimistic Direct Execution to be turned on by default on .NET SDK (#4225). **WARNING:** This is breaking change for GA
+  > Note: Starting Version 3.38.0, the .NET SDK enables the ODE feature by default. This can potentially cause a new type of continuation token to be generated. Such a token is not recognized by the older SDKs by design and this could result in a Malformed Continuation Token Exception.
```
Comment (ealsur) [inline changelog.md:37]: Might want to move this note under the version title, otherwise it might make the list weird

File: changelog.md [hunk 7]
```diff
@@ -13,6 +13,38 @@ Preview features are treated as a separate branch and will not be included in th
...
+  > If you have a scenario where tokens generated from the newer SDKs are used by an older SDK, we recommend a 2 step approach to upgrade:
+  > - Upgrade to the new SDK and disable ODE, both together as part of a single deployment. Wait for all nodes to upgrade.
+  >    - In order to disable ODE, set EnableOptimisticDirectExecution to false in the QueryRequestOptions. 
+  > - Enable ODE as part of second deployment for all nodes.
+- [4251](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4251) Emulator : Adds support for flag in connection string to ignore SSL check (#4251)
+- [4279](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4279) Region Availability: Adds Spain Central and Mexico Central Regions For Public Usage (#4279)
+- [4283](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4283) CosmosClientOptions: Adds validation to check DisableServerCertificateValidation and ServerCertificateCustomValidationCallback are set together (#4283)
```
Comment (ealsur) [inline changelog.md:44]: Remove? This was a second PR after the Emulator one to address a comment that was left unresolved

File: changelog.md (hunk 6 above)
Comment (kundadebdatta) [reply changelog.md]: Refactored.

File: changelog.md (hunk 7 above)
Comment (kundadebdatta) [reply changelog.md]: Removed.

**Review by ealsur — DISMISSED** [review]

**Review by NaluTripician — DISMISSED** [review]

File: changelog.md [hunk 8]
```diff
@@ -13,6 +13,37 @@ Preview features are treated as a separate branch and will not be included in th
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+### <a name="3.39.0-preview.0"/> [3.39.0-preview.0](https://www.nuget.org/packages/Microsoft.Azure.Cosmos/3.39.0-preview.0) - 2024-01-31
+
+#### Added
+- [4138](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4138) Query: Adds support for LINQ Custom Serializer (#4138)
+
+### <a name="3.38.0"/> [3.38.0](https://www.nuget.org/packages/Microsoft.Azure.Cosmos/3.38.0) - 2024-01-31
+  > Note: Starting Version `3.38.0`, the .NET SDK enables the ODE feature by default. This can potentially cause a new type of continuation token to be generated. Such a token is not recognized by the older SDKs by design and this could result in a Malformed Continuation Token Exception.
```
[the remark this answers is not in the document]
Comment (kundadebdatta) [reply changelog.md]: Agree - Looks more clear. Refactored.

**Review by FabianMeiswinkel — DISMISSED** [review]
LGTM

**Review by ealsur — DISMISSED** [review]
Please wait for @kirankumarkolli in case there is anything missing

File: changelog.md [hunk 9]
```diff
@@ -13,6 +13,37 @@ Preview features are treated as a separate branch and will not be included in th
...
+  >    - In order to disable ODE, set EnableOptimisticDirectExecution to false in the QueryRequestOptions.
+  > - Enable ODE as part of second deployment for all nodes.
+
+#### Fixed
+- [4205](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4205) ClientRetryPolicy: Fixes Metadata Requests Retry Policy (#4205)
+- [4220](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4220) Change Feed Processor: Fixes disposal of unused CancellationTokenSource (#4220)
+- [4229](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4229) GatewayClientStore: Fixes an issue with dealing with invalid JSON HTTP responses (#4229)
+- [4260](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4260) Query: Fixes LINQ Translation of SqlNullLiteral Values (#4260)
+- [4276](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4276) Change Feed Processor: Fixes LeaseLostException on Notifications API for Renewer (#4276)
+- [4291](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4291) Documentation: Fixes DeleteItem API will not return context in response (#4291)
```
Comment (kirankumarkolli) [inline changelog.md:34]: Not required.

**Review by kirankumarkolli — DISMISSED** [review]

File: changelog.md (hunk 9 above)
Comment (kundadebdatta) [reply changelog.md]: Removed.

**Review by kirankumarkolli — APPROVED** [review]

**Review by ealsur — APPROVED** [review]

**Pull request merged**